src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We have seen in the Buffer creation section that it is required to provide the buffer location during the buffer creation. XRT provides an API ``xrtKernelArgGroupId`` (C++: ``xrt::kernel::group_id``) that returns the bank index (ID) of a specific argument of the kernel. This ID is used as the last argument of ``xclAllocBO`` (in C++ with ``xrt::bo`` constructor) API to create the buffer on the same memory bank. 
+
+
+Let us review the example below where the buffer is allocated for the kernel's first (argument index 0) by using this API
+
+.. code:: c
+      :number-lines: 39
+           
+           xrtMemoryGroup idx_0 = xrtKernelArgGroupId(kernel, 0); // bank index of 0th argument
+           xrtBufferHandle a = xrtBOAlloc(device, data_size*sizeof(int), XCL_BO_FLAGS_NONE, idx_0);
+
+
+.. code:: c++
+      :number-lines: 15
+                       
+           auto input_buffer = xrt::bo(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, kernel.group_id(0));
+
+
+
+The API fails if the kernel bank index is ambiguous. For example, the kernel has multiple CU with different connectivity for that argument. In those cases, it is required to create a kernel object/handle with specific a CU (or group of CUs with identical connectivity). 
+
+
+   
+Reading and write CU mapped registers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To read and write from the AXI-Lite register space corresponding to a CU, the CU must be opened in exclusive mode (in shared mode, multiple processes can access the CU's address space, hence it is unsafe if they are trying to access/change registers at the same time leading to a potential race behavior). The required APIs for kernel register read and write are
+  
+    - ``xrtKernelReadRegister`` (C++: Method ``xrt::kernel::read_register``)
+    - ``xrtKernelWriteRegiste`` (C++: Method ``xrt::kernel::write_register``)
+
+.. code:: c
+      :number-lines: 35
+         
+           int read_data; 
+           int write_data = 7; 
+              
+           xrtKernelHandle kernel = xrtPLKernelOpenExclusive(device, xclbin_uuid, "foo:{foo_1}");
+              
+           xrtKernelReadRegister(kernel,READ_OFFSET,&read_data);
+           xrtKernelWriteRegister(kernel,WRITE_OFFSET,write_data); 
+              
+           xrtKernelClose(kernel);
+
+
+In the above code block
+
+              - The compute unit named "foo_1" (name syntax: "kernel_name:{cu_name}") is opened exclusively.
+              - The Register Read/Write operation is performed. 
+              - Closed the kernel
+              
+**C++**: The equivalent C++ API example
+
+.. code:: c
+      :number-lines: 35
+       
+           int read_data; 
+           int write_data = 7; 
+              
+           auto krnl = xrt::kernel(device, "foo:{foo_1}", xclbin_uuid, true); 
+
+           read_data = kernel.read_register(READ_OFFSET);
+           kernel.write_register(WRITE_OFFSET,write_data); 
```

Comment:
Make offset optional when 0?
So prefer
```suggestion
           kernel.write_register(write_data, WRITE_OFFSET); 
```

Reply:
Again, please review the APIs and suggest changes to those rather than this PR.
This is a valid point and I agree.

Reply:
implementation specific, tracking it through CR, nothing needs to be done for this PR

Reply:
> Again, please review the APIs and suggest changes to those rather than this PR.
> This is a valid point and I agree.

Sure. Where these suggestions have to be done exactly?